ggml/src/ggml-cuda/cpy-utils.cuh

```diff
@@ -2,26 +2,20 @@
 
 #include "ggml-common.h"
 
-static __device__ __forceinline__ void convert_f32_f32(const float * src, float * dst) {
-    *dst = *src;
+template<typename src_t, typename dst_t>
+static __device__ __forceinline__ void convert_to_flt(const src_t * src, dst_t * dst) {
```

Comment:
Add a check for whether `src_t` and `dst_t` are the same, don't do a cast to float in that case (also covers FP16 -> FP16).